{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dash\n",
    "from dash import html,dcc\n",
    "from matplotlib.colors import colorConverter\n",
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "from dash.dependencies import Input, Output\n",
    "import pickle\n",
    "from numpy import mean\n",
    "#from functions import mittelwertrechner\n",
    "\n",
    "#setup\n",
    "with open(\"metrics.pickle\", 'rb') as handle:\n",
    "    testdatei= pickle.load(handle) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\maxri\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python310\\site-packages\\numpy\\core\\fromnumeric.py:3430: FutureWarning: In a future version, DataFrame.mean(axis=None) will return a scalar mean over the entire DataFrame. To retain the old behavior, use 'frame.mean(axis=0)' or just 'frame.mean()'\n",
      "  return mean(axis=axis, dtype=dtype, out=out, **kwargs)\n",
      "C:\\Users\\maxri\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python310\\site-packages\\numpy\\core\\fromnumeric.py:3430: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "  return mean(axis=axis, dtype=dtype, out=out, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Estimated Number of Cells</th>\n",
       "      <th>Mean Reads per Cell</th>\n",
       "      <th>Median Genes per Cell</th>\n",
       "      <th>Number of Reads</th>\n",
       "      <th>Valid Barcodes</th>\n",
       "      <th>Sequencing Saturation</th>\n",
       "      <th>Q30 Bases in Barcode</th>\n",
       "      <th>Q30 Bases in RNA Read</th>\n",
       "      <th>Q30 Bases in UMI</th>\n",
       "      <th>Reads Mapped to Genome</th>\n",
       "      <th>Reads Mapped Confidently to Genome</th>\n",
       "      <th>Reads Mapped Confidently to Intergenic Regions</th>\n",
       "      <th>Reads Mapped Confidently to Intronic Regions</th>\n",
       "      <th>Reads Mapped Confidently to Exonic Regions</th>\n",
       "      <th>Reads Mapped Confidently to Transcriptome</th>\n",
       "      <th>Reads Mapped Antisense to Gene</th>\n",
       "      <th>Fraction Reads in Cells</th>\n",
       "      <th>Total Genes Detected</th>\n",
       "      <th>Median UMI Counts per Cell</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BfxProjekt</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>bfx1932</th>\n",
       "      <td>2826.500000</td>\n",
       "      <td>115228.250000</td>\n",
       "      <td>2615.125</td>\n",
       "      <td>2.746095e+08</td>\n",
       "      <td>0.971625</td>\n",
       "      <td>0.784500</td>\n",
       "      <td>0.958625</td>\n",
       "      <td>0.934250</td>\n",
       "      <td>0.9545</td>\n",
       "      <td>0.960000</td>\n",
       "      <td>0.903625</td>\n",
       "      <td>0.061750</td>\n",
       "      <td>0.596000</td>\n",
       "      <td>0.245750</td>\n",
       "      <td>0.507375</td>\n",
       "      <td>0.318625</td>\n",
       "      <td>0.869625</td>\n",
       "      <td>29739.875000</td>\n",
       "      <td>5952.125000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bfx1958</th>\n",
       "      <td>1322.875000</td>\n",
       "      <td>151684.625000</td>\n",
       "      <td>3817.250</td>\n",
       "      <td>1.703632e+08</td>\n",
       "      <td>0.975375</td>\n",
       "      <td>0.745125</td>\n",
       "      <td>0.955000</td>\n",
       "      <td>0.929500</td>\n",
       "      <td>0.9495</td>\n",
       "      <td>0.959125</td>\n",
       "      <td>0.919750</td>\n",
       "      <td>0.038625</td>\n",
       "      <td>0.612375</td>\n",
       "      <td>0.268625</td>\n",
       "      <td>0.592875</td>\n",
       "      <td>0.269000</td>\n",
       "      <td>0.910875</td>\n",
       "      <td>26544.125000</td>\n",
       "      <td>13019.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bfx1966</th>\n",
       "      <td>5813.055556</td>\n",
       "      <td>74057.444444</td>\n",
       "      <td>2421.500</td>\n",
       "      <td>3.000849e+08</td>\n",
       "      <td>0.966222</td>\n",
       "      <td>0.701167</td>\n",
       "      <td>0.942944</td>\n",
       "      <td>0.921889</td>\n",
       "      <td>0.9345</td>\n",
       "      <td>0.948278</td>\n",
       "      <td>0.888500</td>\n",
       "      <td>0.043556</td>\n",
       "      <td>0.621833</td>\n",
       "      <td>0.223111</td>\n",
       "      <td>0.539778</td>\n",
       "      <td>0.288056</td>\n",
       "      <td>0.907556</td>\n",
       "      <td>30216.111111</td>\n",
       "      <td>5916.055556</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Estimated Number of Cells  Mean Reads per Cell  \\\n",
       "BfxProjekt                                                   \n",
       "bfx1932                   2826.500000        115228.250000   \n",
       "bfx1958                   1322.875000        151684.625000   \n",
       "bfx1966                   5813.055556         74057.444444   \n",
       "\n",
       "            Median Genes per Cell  Number of Reads  Valid Barcodes  \\\n",
       "BfxProjekt                                                           \n",
       "bfx1932                  2615.125     2.746095e+08        0.971625   \n",
       "bfx1958                  3817.250     1.703632e+08        0.975375   \n",
       "bfx1966                  2421.500     3.000849e+08        0.966222   \n",
       "\n",
       "            Sequencing Saturation  Q30 Bases in Barcode  \\\n",
       "BfxProjekt                                                \n",
       "bfx1932                  0.784500              0.958625   \n",
       "bfx1958                  0.745125              0.955000   \n",
       "bfx1966                  0.701167              0.942944   \n",
       "\n",
       "            Q30 Bases in RNA Read  Q30 Bases in UMI  Reads Mapped to Genome  \\\n",
       "BfxProjekt                                                                    \n",
       "bfx1932                  0.934250            0.9545                0.960000   \n",
       "bfx1958                  0.929500            0.9495                0.959125   \n",
       "bfx1966                  0.921889            0.9345                0.948278   \n",
       "\n",
       "            Reads Mapped Confidently to Genome  \\\n",
       "BfxProjekt                                       \n",
       "bfx1932                               0.903625   \n",
       "bfx1958                               0.919750   \n",
       "bfx1966                               0.888500   \n",
       "\n",
       "            Reads Mapped Confidently to Intergenic Regions  \\\n",
       "BfxProjekt                                                   \n",
       "bfx1932                                           0.061750   \n",
       "bfx1958                                           0.038625   \n",
       "bfx1966                                           0.043556   \n",
       "\n",
       "            Reads Mapped Confidently to Intronic Regions  \\\n",
       "BfxProjekt                                                 \n",
       "bfx1932                                         0.596000   \n",
       "bfx1958                                         0.612375   \n",
       "bfx1966                                         0.621833   \n",
       "\n",
       "            Reads Mapped Confidently to Exonic Regions  \\\n",
       "BfxProjekt                                               \n",
       "bfx1932                                       0.245750   \n",
       "bfx1958                                       0.268625   \n",
       "bfx1966                                       0.223111   \n",
       "\n",
       "            Reads Mapped Confidently to Transcriptome  \\\n",
       "BfxProjekt                                              \n",
       "bfx1932                                      0.507375   \n",
       "bfx1958                                      0.592875   \n",
       "bfx1966                                      0.539778   \n",
       "\n",
       "            Reads Mapped Antisense to Gene  Fraction Reads in Cells  \\\n",
       "BfxProjekt                                                            \n",
       "bfx1932                           0.318625                 0.869625   \n",
       "bfx1958                           0.269000                 0.910875   \n",
       "bfx1966                           0.288056                 0.907556   \n",
       "\n",
       "            Total Genes Detected  Median UMI Counts per Cell  \n",
       "BfxProjekt                                                    \n",
       "bfx1932             29739.875000                 5952.125000  \n",
       "bfx1958             26544.125000                13019.000000  \n",
       "bfx1966             30216.111111                 5916.055556  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def mittelwert(table, Kriterium):\n",
    "    wert = table.groupby(Kriterium).apply(mean)\n",
    "    return wert\n",
    "\n",
    "mittelwert(table=testdatei, Kriterium=\"BfxProjekt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.7 64-bit (microsoft store)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "353a108ecf55237c5a88d1d9ef97d26e6ce730c5a675945b3088fe78a2ead665"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
